CRM/Custom/Page/CustomDataTrait.php

```diff
@@ -35,15 +35,21 @@ trait CRM_Custom_Page_CustomDataTrait {
    *
    */
   protected function getCustomDataFieldsForEntityDisplay(string $entity, int $id): array {
-    $values = civicrm_api4($entity, 'get', [
-      'select' => [
-        'custom.*',
-      ],
-      'where' => [
-        ['id', '=', $id],
-      ],
-      'checkPermissions' => TRUE,
-    ])->single();
+    try {
+      $values = civicrm_api4($entity, 'get', [
+        'select' => [
+          'custom.*',
+        ],
+        'where' => [
+          ['id', '=', $id],
+        ],
+        'checkPermissions' => TRUE,
+      ])->single();
```

Comment:
Is `single()` still the correct method to call? It is meant for when you are sure that there is only a single result, but that you catch the exception seems to imply that this isn't always the case.

Reply:
The problem is that sometimes the array returned is empty, presumably due to a failed ACL check, not because there is actually no Address or Email. This leads the call to fail, which is now captured by the exception.

Reply:
@Sjord I feel like to keep the change as minimal as possible it makes sense to keep the code as is and just catch the exception as needed.